modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp

```diff
@@ -0,0 +1,311 @@
+    {
+        virtual void unpackKernel(ade::Graph            &graph,
+                                  const ade::NodeHandle &op_node,
+                                  const cv::GKernelImpl &impl) override
+        {
+            GPlaidMLModel gm(graph);
+            auto plaidml_impl = cv::util::any_cast<cv::GPlaidMLKernel>(impl.opaque);
+            gm.metadata(op_node).set(cv::gimpl::PlaidMLUnit{plaidml_impl});
+        }
+
+        virtual EPtr compile(const ade::Graph &graph,
+                             const cv::GCompileArgs &,
+                             const std::vector<ade::NodeHandle> &nodes,
+                             const std::vector<cv::gimpl::Data>& ins_data,
+                             const std::vector<cv::gimpl::Data>& outs_data) const override
+        {
+            return EPtr{new cv::gimpl::GPlaidMLExecutable(graph, nodes, ins_data, outs_data)};
+        }
+   };
+}
+
+cv::gapi::GBackend cv::gapi::plaidml::backend()
+{
+    static cv::gapi::GBackend this_backend(std::make_shared<GPlaidMLBackendImpl>());
+    return this_backend;
+}
+
+void cv::gimpl::GPlaidMLExecutable::initBuffers(const std::vector<cv::gimpl::Data>& data,
+                                                std::vector<plaidml::exec::Binding>& bindings)
+{
+
+    // NB: This is necessary because we keep a pointer to bindings elements to buffer_map
+    // In order to them to remain valid it's required to prevant reallocation
+    bindings.reserve(data.size());
+    for (const auto& d : data)
+    {
+        GAPI_Assert(d.shape == GShape::GMAT &&
+                    "Now PlaidML backend supported only cv::GMat's");
+
+        const auto& desc = cv::util::get<cv::GMatDesc>(d.meta);
+
+        auto placeholder = plaidml::edsl::Placeholder(
+                           cv::util::plaidml::depth_from_ocv(desc.depth),
+                           {desc.size.width, desc.size.height, desc.chan});
+
+        const auto& shape = placeholder.shape();
+        plaidml::TensorShape tshape(shape.dtype(), shape.int_dims());
+        plaidml::Buffer buffer(device_id_, tshape);
+
+        bindings.push_back(plaidml::exec::Binding{std::move(placeholder),
+                                                  std::move(buffer)});
+
+        auto& tensor_map = m_res.slot<plaidml::edsl::Tensor>();
+        // FIXME Avoid Copy here !!!
+        tensor_map.emplace(d.rc, bindings.back().tensor);
+
+        auto& buffer_map = m_res.slot<plaidml::Buffer*>();
+        buffer_map.emplace(d.rc, &(bindings.back().buffer));
+    }
+}
+
+void cv::gimpl::GPlaidMLExecutable::initConfig()
+{
+    auto read_var_from_env = [](const char* env)
+    {
+        const char* raw = std::getenv(env);
+        if (!raw)
+        {
+            cv::util::throw_error(std::runtime_error(std::string(env) + " is't set"));
+        }
+
+        return std::string(raw);
+    };
+
+    device_id_ = read_var_from_env("PLAIDML_DEVICE");
+    target_id_ = read_var_from_env("PLAIDML_TARGET");
+}
```

Comment:
I'd make it graph compilation arguments. Why not?

Reply:
I thought. Is it possible with this API to run this on CI ? You do not know the name of the device. 
But if you specify this through the environment, you can get the name of the device from outside the code.

Reply:
I'd make this the default solution and add the compile argument as optional ? What do you think ?

Reply:
We need to avoid relying on the environment variables in our library.

Instead, you still could read from the environment variables in your test and pass those as compile args down to G-API.

Reply:
Fixed